### PR TITLE
Add ElasticSearch Connection Doc

### PR DESCRIPTION
**Pull request by bryce-lewis**
Added a connection doc for the ElasticSearch connection. Added links inside the ElasticSearch module doc strings for user to have access to the ElasticSearch connection doc.

### CODE REVIEW

**Comment by kaxil** [comment]
Pre-commit checks are failing, can you install and re-run pre-commit checks on your machine please:

https://github.com/apache/airflow/pull/16436/checks?check_run_id=2822574119#step:10:273

File: docs/apache-airflow-providers-elasticsearch/connections/elasticsearch.rst [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+it using URI syntax.
+
+Note that all components of the URI should be URL-encoded.
+
+For example:
+
+.. code-block:: bash
+
+    export AIRFLOW_CONN_ELASTICSEARCH_DEFAULT='elasticsearch://elasticsearchlogin:elasticsearchpassword@elastic.co:80/http'
+
```
Comment (kaxil) [inline docs/apache-airflow-providers-elasticsearch/connections/elasticsearch.rst]:
```suggestion
```

File: docs/apache-airflow-providers-elasticsearch/connections/elasticsearch.rst [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+Port
+    Specify the Elasticsearch port for the initial connection
+
+Scheme
+    Specify the schema for the Elasticsearch API. `http` by default
+
+Extra (Optional)
+    Specify the extra parameters (as json dictionary) that can be used in Azure connection. 
+    The following parameters are all optional:
+
```
Comment (kaxil) [inline docs/apache-airflow-providers-elasticsearch/connections/elasticsearch.rst]:
```suggestion
    Specify the extra parameters (as json dictionary) that can be used in Azure connection.
    The following parameters are all optional:

```

File: docs/apache-airflow-providers-elasticsearch/connections/elasticsearch.rst [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+Some hooks and operators related to ElasticSearch use elasticsearch_default by default.
+
+Configuring the Connection
+--------------------------
+
+User 
+    Specify the login used for the initial connection
+
+Password
```
Comment (kaxil) [inline docs/apache-airflow-providers-elasticsearch/connections/elasticsearch.rst]:
```suggestion

User
    Specify the login used for the initial connection

Password
```

File: airflow/providers/elasticsearch/hooks/elasticsearch.py [hunk 4]
```diff
@@ -30,7 +30,8 @@ class ElasticsearchHook(DbApiHook):
 
     This hook uses the Elasticsearch conn_id.
 
-    :param elasticsearch_conn_id: The Airflow connection used for Elasticsearch credentials.
+    :param elasticsearch_conn_id: The :ref:`ElasticSearch connection id <howto/connection:elasticsearch>` 
+        used for Elasticsearch credentials.
```
Comment (kaxil) [inline airflow/providers/elasticsearch/hooks/elasticsearch.py]:
```suggestion
    :param elasticsearch_conn_id: The :ref:`ElasticSearch connection id <howto/connection:elasticsearch>`
        used for Elasticsearch credentials.
```

**Review by potiuk — APPROVED** [review]

**Review by 0ex-d — APPROVED** [review]

**Pull request merged**